src/foam/nanos/http/SessionWebAgent.java

```diff
@@ -44,12 +45,34 @@ public void execute(X x) {
     HttpServletResponse resp       = x.get(HttpServletResponse.class);
 
     try {
-      // check session id
-      String sessionId = req.getParameter("sessionId");
+      String sessionId = "";
+
+      // check for session id in cookie
+      Cookie[] cookies = req.getCookies();
+      for ( Cookie c : cookies ) {
+        if ( c.getName().equals("sessionId") ) {
+          sessionId = c.getValue();
```

Comment:
Super minor, but you could add a `break` after this line since we don't need to loop over the rest of the cookies.

Reply:
Updated